aiida_siesta/docs/workflows/converger.rst

```diff
@@ -0,0 +1,92 @@
+SIESTA Converger workflow
++++++++++++++++++++++++++++++++++
+
+Description
+-----------
+
+The **SiestaConvereger** is a tool to facilitate convergence tests with Siesta.
+It extends the **SiestaIterator** to accept a target quantity that is checked
+after each step to evaluate whether convergence has been reached or not.
+The convergence check just consists in calculating the difference in the target quantity 
+between the present step and the step before and comparing it with a threshold value
+passed by the user in input.
+An example on the use of the **SiestaConverger** is
+/aiida_siesta/examples/workflows/example_convergence.py.
+
+
+Supported Siesta versions
+-------------------------
+
+At least 4.0.1 of the 4.0 series, and 4.1-b3 of the 4.1 series, which
+can be found in the development platform
+(https://gitlab.com/siesta-project/siesta).
+
+Inputs
+------
+
+All the **SiestaIterator** inputs are as well inputs of the **SiestaConvereger**,
+they are described in the corresponding
+:ref:`documentation <siesta-iterator-inputs>`.
+Additional inputs are:
+
+* **target**, class :py:class:`Str  <aiida.orm.Str>`, *Optional*
+
+  The parameter the user wants to track in order to check if convergence has been reached.
+  All the quantities returned in the `output_parameters` dictionary of the **SiestaBaseWorkChain**
+  are accepted for this scope, excluding keys that don't have a `float` or `int` as a value.
+  Typical values are the Kohn-Sham
+  (`E_KS`), Free (`FreeE`), Band (`Ebs`), and Fermi (`E_Fermi`)
+  energies, and the total spin (`stot`); however the user might also think to converge
+  calculations-time related quantities.
+
+  The `E_KS` is the default value.
+
+
+.. |br| raw:: html
+
+    <br />
+
+* **threshold**, class :py:class:`Float <aiida.orm.Float>`, *Optional*
+
+  The maximum difference between two consecutive steps to consider that convergence is reached.
+  Default is `Float(0.01)`.
+
+Outputs
+-------
+
+The following outputs are returned:
+
+* **converged** :py:class:`Bool <aiida.orm.Bool>`
+
+  Returning `True` or `False`, whether the target has converged or not.
+
+.. |br| raw:: html
+
+    <br />
+
+* **converged_target_value** :py:class:`Float <aiida.orm.Float>`
+
+  The value of the target when the convergence has been reached, or in the last step of
+  the iteration.
+
+.. |br| raw:: html
+
+    <br />
```

Comment:
On the same topic, why is `iterate_over` optional in your implementation?

Reply:
Hmm I think at the beggining was to allow nested iterators. For example the `SequentialConverger` would not work if `iterate_over` is required because the `converger_inputs` namespace would ask for it. But then I decided to exclude it when I expose the inputs, so this problem is not there anymore. I think it can be made required without any problem at the moment.

Reply:
Ok. I made the change. To be honest I didn't test the SequentialConverger, but I took note for the future. Thanks!

Reply:
Hahahah running the `SiestaSequentialConverger` is the best test for the implementation, because it uses almost all aspects of it. And if you already have the setup to test `SiestaConverger`, testing `SiestaSequentialConverger` is dead easy, I invite you to do it :)

Reply:
I checked, all good. I actually added an extra example specific for the `SiestaSequentialConverger`, so it's easier. However before documenting and promoting this WorkChain, we need to discuss the API. I don't like so much the need of the extra level of `converger_inputs` dictionary. But now let's give our-self a break. Next week. See ya!